common/src/primitives/id/mod.rs

```diff
@@ -66,26 +69,32 @@ impl<'de> serde::Deserialize<'de> for H256 {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Encode, Decode, serde::Serialize, serde::Deserialize)]
+#[derive(PartialEq, Eq, Encode, Decode, serde::Serialize, serde::Deserialize)]
 #[serde(transparent)]
-pub struct Id<T: ?Sized> {
+pub struct Id<T> {
```

Comment:
Why has `?Sized` been removed?

Reply:
I actually never understood why it's there in the first place. We have no intention to support unsized trait objects and it's actually never used.

Reply:
It's just an unnecessary restriction. Putting the `Sized` constraint there suggests the `T` type is going to be stored or manipulated as a value somewhere within the `Id`, which it is not since it only occurs in `PhantomData`

Reply:
We discussed, and agreed that there's kind of difference in the way we see this as `?Sized` can be seen as a restriction and also an extension of possible types. We'll keep it as is as it's not a big deal right now. If the need comes to change it back, we'll do it then.